src/service_control_client_impl_quota_test.cc

```diff
@@ -234,6 +237,55 @@ TEST_F(ServiceControlClientImplQuotaTest,
   EXPECT_EQ(stat.send_quotas_in_flight, 10);
 }
 
+TEST_F(ServiceControlClientImplQuotaTest, TestCachedQuotaRefreshGotHTTPError) {
+  // callback  returns the negative response
+  mock_quota_transport_.quota_response_ = &error_quota_response1_;
+  mock_quota_transport_.done_status_ = Status::OK;
+
+  EXPECT_CALL(mock_quota_transport_, Quota(_, _, _))
+      .WillRepeatedly(
+          Invoke(&mock_quota_transport_,
+                 &MockQuotaTransport::AllocateQuotaWithInplaceCallback));
+
+  Status done_status = Status::UNKNOWN;
+  AllocateQuotaResponse quota_response;
+
+	// Initial Quota
+  cached_client_->Quota(
+      quota_request1_, &quota_response,
+      [&done_status](Status status) { done_status = status; });
+  EXPECT_EQ(done_status, Status::OK);
+
+  // callback returns HTTP error
+  mock_quota_transport_.done_status_ = Status::CANCELLED;
+
+  // wait for the next quota refresh
+  std::this_thread::sleep_for(std::chrono::milliseconds(600));
+
+  // triggers quota cache refresh and returns previously cached response.
+  // callback function updates the cache with dummy positive response
+  // for failed open
+  cached_client_->Quota(quota_request1_, &quota_response,
+      [&done_status](Status status) { done_status = status; });
+  EXPECT_EQ(quota_response.allocate_errors_size(), 1);
```

Comment:
did not understand why this one fail and next one is ok.  Could you add some comment?